internal/backend/local/test.go

```diff
@@ -1024,6 +1031,32 @@ func (runner *TestFileRunner) GetVariables(config *configs.Config, run *modulete
 		diags = diags.Append(valueDiags)
 	}
 
+	// We need to also process all global variables for tests.
+	// If we run "terrafrom test" from the testing directory itself,
+	// the filepath.Dir(file.Name) is equal to "." so we need to extend the logic
+	if filepath.Dir(file.Name) == runner.Suite.TestingDirectory || filepath.Dir(file.Name) == "."{
```

Comment:
```suggestion
	if filepath.Dir(file.Name) == runner.Suite.TestingDirectory {
```

We don't run the test command directly from the testing directory, so no need to check for `.`. I think this will break your test though, so you should move the test file into `./tests`.